### DISCUSSION TITLE
Order/synchronism of postTargets

### DISCUSSION THREAD

**Discussion by eahenke** [jscutlery/semver#781] [Q&A] [answered]
Apologies in advance if this is clear from the code - I'm really unfamiliar with rxjs. I'm wondering if the `postTargets` array is run sequentially/synchronously, ie - if I have `postTargets: [`a`, `b`, `c`]`, can I rely on `a` running **and completing** before `b` runs, and so forth? Or is it more of a `Promise.all` situation? I ask because as part of my publishing workflow I'd like to version, then copy the new `package.json` over to the dist folder, and then run publish, in that order. Thank you!

**Comment by edbzn** [comment, marked as answer]
Hi, postTargets run sequentially so the order in the array matters, and note that if a target fails semver will exit with an error, skipping any remaining target.

**Reply by eahenke** [reply]
Thank you!
